release.py

```diff
@@ -177,18 +183,66 @@ def push() -> None:
             raise typer.Exit(result)
 
 
-def check_preconditions() -> None:
+def ensure_all_committed_files() -> Optional[List[str]]:
+    command = ("git", "ls-files", "-m")
+    typer.echo(" ".join(command))
+    result = subprocess.run(command, capture_output=True)
+
+    uncommitted_files = result.stdout.decode("UTF-8").split("\n")[:-1]
+    while len(uncommitted_files) > 0:
+        typer.echo(f"There are some uncommitted files: {uncommitted_files}.")
+        typer.echo("You may commit or stash them in another tab and retry")
+        should_retry = typer.confirm("Only committed changes will be released. Retry ?")
+        if not should_retry:
+            raise typer.Abort()
+        uncommitted_files = result.stdout.decode("UTF-8").split("\n")[:-1]
+
+    typer.echo("All files are committed, proceeding")
+    return True
+
+
+def generate_doc() -> None:
+    command = ("make", "doc")
+    typer.echo(" ".join(command))
+    return subprocess.run(command)
+
+
+def ask_check_preconditions(bump_type) -> None:
+    """
+    Check preconditions that cannot be checked from this script.
+    """
+    current_version, bumped_version = get_current_and_bumped_version(bump_type)
     preconditions = [
-        "Did you commit everything that needs to be in the new version?",
-        f"Did you update the changelog at {CHANGELOG}?",
-        "Did you generate the doc using `make doc`?",
-        "Did you add the new version to the compatibility mapping in the API?",
+        {
+            "message": f"Going to update from version {current_version} to {bumped_version}"
+        },
+        {
+            "message": f"Did you add {bumped_version} to the compatibility mapping in the API?"
+            + " NB: you can edit it in another tab and press Y after that"
```

Comment:
```suggestion
            + " You can edit it in another tab and press Y after that"
```
IMO, we can be more formal

Reply:
why not